File: applicatie/src/main/java/com/vdzon/administratie/rubriceren/rubriceerRegels/RubriceerHelper.java
package com.vdzon.administratie.rubriceren.rubriceerRegels;

import com.vdzon.administratie.model.Gebruiker;

import java.util.Comparator;

public class RubriceerHelper {
    static int findNextRekeningNummer(Gebruiker gebruiker) {
        return 1 + gebruiker.getDefaultAdministratie().getRekeningen().stream().map(rekening -> Integer.parseInt(rekening.getRekeningNummer())).max(Comparator.naturalOrder()).orElse(1000);
    }

}

File: applicatie/src/main/java/com/vdzon/administratie/rubriceren/rubriceerRegels/RubriceerHelper.kt
package com.vdzon.administratie.rubriceren.rubriceerRegels

import com.vdzon.administratie.model.Gebruiker

import java.util.Comparator

open class RubriceerHelper {
    companion object {
        internal fun findNextRekeningNummer(gebruiker: Gebruiker): Int {
            return 1 + (gebruiker.defaultAdministratie.rekeningen.map({ rekening -> Integer.parseInt(rekening.rekeningNummer) }).maxWith(Comparator.naturalOrder<Int>())?:1000)
        }
    }
}
